## Modifying genome_sucking.py
_Date: 03/15/2017_

In [2]:
import pandas as pd
import re
import os

In [4]:
pathcsv="/home/savandara/Documents/Internship/2017_master_internship/curated_insectGenomesInfos.csv"

datatofetch = pd.read_csv(pathcsv)
gIDtofetch = datatofetch['Accession number'].values
len(gIDtofetch)

246

_Date: 03/15/2017_
### Comment:
- I have noticed that our file AllGenomes is not updated because I cannot find all the Accession number listed in gIDtofetch (just 171 on 246) 

$\rightarrow$ Looking for some update of our datafile AllGenome to fix the problem (__not found!__)

$\rightarrow$ __Solution from Leo__: Change all the code, don't need to use a file called AllGenomes, just have to fetch all genomes through their repertorie : ftp://ftp.ncbi.nih.gov/genomes/all/.../.../.../... (with regular expressions)


In [89]:
for i in range(len(gIDtofetch)):
	arg = re.findall("(^[A-Z]{3})_([0-9]{3})([0-9]{3})([0-9]{3})", gIDtofetch[i])[0]
	os.system('rsync --copy-links --recursive --times --verbose rsync://ftp.ncbi.nlm.nih.gov/genomes/all/{}/{}/{}/{}/ ../AllGenomes/'.format(arg[0], arg[1], arg[2], arg[3]))


# Done 
- `sucking_genomeV2.py` works !!